doc/api/test_runner.md

```diff
@@ -0,0 +1,275 @@
+# Test runner
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental
+
+<!-- source_link=lib/test_runner.js -->
+
+The `test_runner` module facilitates the creation of JavaScript tests that
+report results in [TAP][] format. To access it:
+
+```js
+const test = require('test_runner');
+```
+
+Tests created via the `test_runner` module consist of a single function that
+executes either synchronously or asynchronously. Synchronous tests are
+considered passing if they do not throw an exception. Asynchronous tests return
+a `Promise`, and are considered passing if the returned `Promise` does not
+reject. The following example illustrates how tests are written using the
+`test_runner` module.
+
+```js
+const assert = require('assert');
+const test = require('test_runner');
```

Comment:
Not blocking but worth considering: as a new top level module this is semver-major. If it were hung off `assert` instead it would be semver-minor and could be backported.

Reply:
I think in the long term, having the testing API separately makes more sense. If there is a strong enough desire to backport it, the TSC can always vote to do so - especially since this doesn't conflict with anything on npm.

Reply:
I'd note it isn't a breaking change if it only lives under `node:` as well.

Reply:
By using `test_runner` there will also be the `node:test_runner` namespace used, right? Having only `node:test` seems much nicer?!

Reply:
This is now exposed as `node:test` and only works under the `node:` scheme.

Reply:
To clarify:
- Is exposing only to `node:test` and not under `test`, it allows to land this new module for v17 (not as a semver major)?
- Is it planned then for v18 (as it is maybe a breaking change), to also expose it under `test` (not only `node:test`)?

Reply:
> Is exposing only to node:test and not under test

That's correct, and is called out in the docs.

However, we didn't previously have a way to ship `node:`-only modules (at least not that I was aware of). That functionality is added here, but I'm leaving this as semver major because I'm not convinced that adding `node:`-only modules is not a breaking change as evidenced by the required changes to our own test suite.